File: Programing with C#/C# Fundamentals II/Exam/exam-csharp/1 Taks/StrangeLand Numbers.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using System.Numerics;

class StrangeLandNumbers
{
    //private static string[] numbers = new string[] {"f","bIN","oBJEC","mNTRAVL","lPVKNQ","pNWE","hT" };

    private static Dictionary<string, int> numbers = new Dictionary<string, int>();
    private static void InitDic() {
        numbers.Add("f", 0);
        numbers.Add("bIN", 1);
        numbers.Add("oBJEC", 2);
        numbers.Add("mNTRAVL", 3);
        numbers.Add("lPVKNQ", 4);
        numbers.Add("pNWE", 5);
        numbers.Add("hT", 6);
    }
    static void Main(string[] args)
    {
        if (File.Exists("input.txt"))
        {
            Console.SetIn(new StreamReader("input.txt"));
        }
        InitDic();
        string line = Console.ReadLine();

        List<int> resNum = new List<int>();
        StringBuilder buffer = new StringBuilder();
        //int starIndex = 0;
        //buffer.Append(line.Substring(0, 1));
        for (int i = 0; i < line.Length; i++)
        {
            string curChar = line.Substring(i,1);
            buffer.Append(curChar);
            if (numbers.ContainsKey(buffer.ToString()))//found in buffer
            {
                int n = numbers[buffer.ToString()];
                resNum.Add(n);
                buffer.Clear();
            }
       
        }
        BigInteger sum = 0;
        //calculate number
        for (int i = 0; i < resNum.Count; i++)
        {
            
            //BigInteger stepen = BigInteger.Pow(7, resNum.Count - i - 1);
            BigInteger stepen = Pow(resNum.Count - i - 1); // BigInteger.Pow(9, listNum.Count - i - 1);
            BigInteger fnum = (BigInteger)resNum[i];
            sum = BigInteger.Add(sum, (BigInteger.Multiply(fnum, stepen)));
        }
        Console.WriteLine(sum);

    }

    private static BigInteger Pow(int power)
    {
        BigInteger result = 1;
        for (int i = 0; i < power; i++)
        {
            result *= 7;
        }
        return result;
    }
}

File: Programing with C#/C# Fundamentals II/Exam/exam-java/1 StrangeLand Numbers/src/StrangeLandNumbers.java
import java.io.BufferedInputStream;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Scanner;

public class StrangeLandNumbers {

	private static Map<String, Integer> numbers = new HashMap<>();

	public static void main(String[] args) {
		String dir = System.getProperty("user.dir");
		String fileName = dir + "\\input.txt";
		File f = new File(fileName);
		if (f.exists()) {
			try {
				System.setIn(new BufferedInputStream(new FileInputStream(
						fileName)));
				System.setIn(new FileInputStream(fileName));
			} catch (FileNotFoundException e) {
				System.out.println(e.getMessage());
			}
		}
		@SuppressWarnings("resource")
		Scanner sc = new Scanner(System.in);

		InitDic();

		String line = sc.nextLine();

		List<Integer> resNum = new ArrayList<Integer>();
		StringBuilder buffer = new StringBuilder();
		for (int i = 0; i < line.length(); i++) {
			String curChar = line.substring(i, i+1);
			buffer.append(curChar);
			if (numbers.containsKey(buffer.toString())) {
				int n = numbers.get(buffer.toString());
				resNum.add(n);
				buffer = new StringBuilder();
			}

		}
		BigInteger sum = BigInteger.valueOf(0);
		for (int i = 0; i < resNum.size(); i++) {
			BigInteger stepen = Pow(resNum.size() - i - 1);
			BigInteger fnum = BigInteger.valueOf((long) resNum.get(i));
			sum = sum.add(fnum.multiply(stepen));
		}
		System.out.println(sum);
	}

	private static BigInteger Pow(int power) {
		BigInteger result = BigInteger.valueOf(1);
		for (int i = 0; i < power; i++) {
			result = result.multiply(BigInteger.valueOf(7));
		}
		return result;
	}

	private static void InitDic() {
		numbers.put("f", 0);
		numbers.put("bIN", 1);
		numbers.put("oBJEC", 2);
		numbers.put("mNTRAVL", 3);
		numbers.put("lPVKNQ", 4);
		numbers.put("pNWE", 5);
		numbers.put("hT", 6);
	}

}
